upstairs/src/lib.rs

```diff
@@ -1072,28 +1218,28 @@ async fn looper(
             eprintln!("ERROR: {}: proc: {:?}", target, e);
         }
 
-        /*
-         * Do not attempt to reconnect if we were marked inactive.
-         */
-        if !up.is_active() {
-            drop(up_coms.ds_status_tx);
```

Comment:
When I was testing, I saw that we have to do these drops if we want the Upstairs to stop cleanly

Reply:
If a downstairs connects, then disconnects before we go active, then it would exit here and can never re-join (this one took me a while to figure out).   

We may need some other global state to tell an upstairs that it should gracefully shutdown though.

Reply:
:+1: